### ISSUE TITLE
Cannot run gtk template using dotnet run

### ISSUE THREAD

**Issue by Dremor** [GtkSharp/GtkSharp#60]
I tried to create a new gtkapp project using dotnet cli, but I cannot run it, even after a `dotnet restore`.

- MSYS2 and GTK are installed
- Build without problems

the error is the following:
```
$ dotnet run

Unhandled Exception: System.TypeInitializationException: The type initializer for 'Gtk.Application' threw an exception. ---> System.DllNotFoundException: Gtk
   at GLibrary.Load(Library library)
   at Gtk.Application..cctor()
   --- End of inner exception stack trace ---
   at Gtk.Application.Init()
   at NetCore_GTK_Playground.Program.Main(String[] args) in C:\[...]\Git\NetCore-GTK-Playground\Program.cs:line 11
```

**Comment by pedrohex** [comment]
Hi,

My guess is youre missing step 3, checkout :

https://github.com/GtkSharp/GtkSharp/wiki/Installing-Gtk-on-Windows

On 22 Oct 2018, at 13:32, Dremor <notifications@github.com<mailto:notifications@github.com>> wrote:


I tried to create a new gtkapp project using dotnet cli, but I cannot run it, even after a dotnet restore.

  *   MSYS2 and GTK are installed
  *   Build without problems

the error is the following:

$ dotnet run

Unhandled Exception: System.TypeInitializationException: The type initializer for 'Gtk.Application' threw an exception. ---> System.DllNotFoundException: Gtk
   at GLibrary.Load(Library library)
   at Gtk.Application..cctor()
   --- End of inner exception stack trace ---
   at Gtk.Application.Init()
   at NetCore_GTK_Playground.Program.Main(String[] args) in C:\[...]\Git\NetCore-GTK-Playground\Program.cs:line 11

**Comment by cra0zy** [comment]
You need to restart the system after step 3.

**Comment by Dremor** [comment]
Indeed, I forgot to restart. Sorry about that.

**Comment by cra0zy** [comment]
No worries, I updated the instructions :)

**Comment by guy-rouillier** [comment]
Hopefully someone will see updates to this issue, even though it is closed.  I'm at the end of my rope with this problem, running on Ubuntu Mate 19.04 using Snap Store installation of dotnet-sdk-2.2 and code.  I have other installations of the same Ubuntu Mate 19.04, in which I've installed DotNet and Code using the packages provided by the Microsoft site, and that all works.  I'm trying to figure out why the Snap installation doesn't work, so I can pass that info onto Ubuntu to fix.  BTW, in the snap installation, I've generated the basic Console app, and that works fine.  So it's just GtkSharp I can't get going.

I get the same stack trace posted at the top.  I'm not sure in step 3 if I'm supposed to add the path to GtkSharp.dll or the path to libgtk-3.so, so I added them both.  I ran the code under gdb, and here's what a backtrace produces.  Appreciate any help.

`
Thread 1 "dotnet" received signal SIGABRT, Aborted.
0x00007ffff6c6ee97 in raise () from /snap/dotnet-sdk/42/lib/x86_64-linux-gnu/libc.so.6
(gdb) bt
#0  0x00007ffff6c6ee97 in raise () at /snap/dotnet-sdk/42/lib/x86_64-linux-gnu/libc.so.6
#1  0x00007ffff6c70801 in abort () at /snap/dotnet-sdk/42/lib/x86_64-linux-gnu/libc.so.6
#2  0x00007ffff636ea23 in  () at /snap/dotnet-sdk/42/shared/Microsoft.NETCore.App/2.2.6/libcoreclr.so
#3  0x00007ffff636d81b in  () at /snap/dotnet-sdk/42/shared/Microsoft.NETCore.App/2.2.6/libcoreclr.so
#4  0x00007ffff60d6938 in UnwindManagedExceptionPass1(PAL_SEHException&, _CONTEXT*) ()
    at /snap/dotnet-sdk/42/shared/Microsoft.NETCore.App/2.2.6/libcoreclr.so
#5  0x00007ffff60d6ab3 in DispatchManagedException(PAL_SEHException&, bool) ()
    at /snap/dotnet-sdk/42/shared/Microsoft.NETCore.App/2.2.6/libcoreclr.so
#6  0x00007ffff604932d in JIT_GetSharedNonGCStaticBase_Helper ()
    at /snap/dotnet-sdk/42/shared/Microsoft.NETCore.App/2.2.6/libcoreclr.so
#7  0x00007fff7cf24b39 in  ()
#8  0x00007fff5801ef18 in  ()
#9  0x0000000000000000 in  ()
`

**Comment by cra0zy** [comment]
Snap and flatpak are not supported in any shape or form, install dotnet sdk in a traditional manner and it will work fine.

**Comment by guy-rouillier** [comment]
@cra0zy, thanks for the quick reply.  I think I got started down the Snap route because Ubuntu 19.04 has started packaging components of the system as snaps.  So I saw dotnet in the Snap Store and proceeded.  Here is the Snapcraft site promoting snaps for dotnet, including links to Microsoft:

[Snapcraft DotNet](https://snapcraft.io/dotnet-sdk)

I'll send them a note and point out this solution has critical issues.
